src/zarr/core/config.py

```diff
@@ -64,6 +64,11 @@ def reset(self) -> None:
             },
             "buffer": "zarr.core.buffer.cpu.Buffer",
             "ndbuffer": "zarr.core.buffer.cpu.NDBuffer",
+            "v2_dtype_kind_to_default_filters_and_compressor": {
+                "biufcmM": ["zstd"],
+                "U": ["vlen-utf8"],
+                "OSV": ["vlen-bytes"],
+            },
```

Comment:
This is quite a mouthful. Can we iterate a bit on the name of the config value and the keys?

Reply:
👍 to changing these names. I'm afraid I don't really have enough context to suggest better names, but I think `v2_dtype_kind_to_default_filters_and_compressor` should be shorter, and `bitfuncM`, `U` and `OSV` should be a bit more descriptive (maybe one or two words instead of acroynms)

Reply:
What about?
```
            "v2_default_compressor": {
                "numeric": ["zstd"],
                "str": ["vlen-utf8"],
                "bytes": ["vlen-bytes"],
            },
```

Reply:
I agree with your suggestion. What do you think about `v2_default_compressors` (plural) as one could want multiple default filters and compressors?

Reply:
I think we should keep it as it is in v2. So 
```
            "v2_default_compressor": {
                "numeric": "zstd",
                "string": "vlen-utf8",
                "bytes": "vlen-bytes",
            },
```